src/getMigrateWidgets.ts

```diff
@@ -44,22 +48,53 @@ export const getMigrateWidgets =
       contentServer.children?.ui.children?.widgets.children?.structure!;
     const filesAncestry = _getFilesAncestry(widgetsStructure);
 
+    const migrateWidget = produce(callback);
+
     for (const fileId in widgetsContent) {
       let migratedWidget;
       const { entry } = widgetsContent[fileId];
       const widget = JSON.parse(entry.content);
 
+      const folderName = filesAncestry[fileId].map(({ name }) => name);
+      const folderId = filesAncestry[fileId].map(({ id }) => id);
+      const metadata = _getMetaData(widgetsStructure, folderId, fileId);
+
+      if (keysOfWidgetPluginsToRemove?.includes(widget.key)) {
+        // The widget's plugin key is flagged for removal.
+        // Remove the widget instead of migrating it.
+        counters.widgets.removed++;
+        _addErrorToReport(errorReport, {
+          folderName,
+          folderId,
+          contentType: "widgets",
+          fileErrorReport: {
+            error: _serializeError(
+              new WidgetFlaggedForRemovalError(widget.key),
+              { doesReportIncludeStacks },
+            ),
+          },
+          fileId,
+          name: metadata.name!,
+        });
+        delete widgetsContent[fileId];
+        const parentFolder = folderId.reduce(
+          (acc, id) => acc.children![id],
+          widgetsStructure,
+        );
+        delete parentFolder.children![fileId];
+        continue;
```

Comment:
Out of curiosity, why using this instead of having the rest of the migration in an `else` statement?
I admit that I'm not very familiar with `continue`

Reply:
It's just a coding style preference, but it's one that we try to follow in the monorepo: avoid nesting, and deal with edge cases with early returns (or `continue`/`break`, etc.) as soon as possible.

Reply:
Okay, noted!